lib/environment.js

```diff
@@ -0,0 +1,109 @@
+/* jshint browser: true */
+/* global chrome */
+/**
+ * Abstracts the Environment away into a common object shared between all 
+ * environment types. This allows testing mocks to ignore any 
+ * environment-specific actions.
+ *
+ * @class
+ * @classdesc A wrapper for the environment the client library is running in.
+ */
+function Environment() {
```

Comment:
The major functions preventing testing go in the Environment, but I'm unclear as to whether 1) I'm setting the correct convention here and 2) whether all environment related things should end up here. That could be a lot of things, but it would be worth doing that to maintain consistency. But then we should really talk about 1) so I can get a better feel for what I'm doing with node. Also, it'd be cool to be able to auto generate some of the functions, since they are all so similar.